components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -133,7 +133,7 @@ class GeckoEngineSession(
     override fun saveState(): Map<String, Any> = runBlocking {
         val stateMap = CompletableDeferred<Map<String, Any>>()
 
-        ThreadUtils.sGeckoHandler.post {
+        ThreadUtils.postToBackgroundThread {
             geckoSession.saveState().then({ state ->
```

Comment:
Same concern as @pocmo. I remember this not working on a background thread. Does it now?

Reply:
Yup you are both right! See my update in `ThreadUtils`